core/http_client.go

```diff
@@ -104,16 +89,18 @@ func (w httpRequestDoerAdapter) Do(req *http.Request) (*http.Response, error) {
 	return w.fn(req)
 }
 
-// CreateHTTPClient creates a new HTTP client with the given client configuration.
+// CreateHTTPInternalClient creates a new HTTP client with the given client configuration.
+// This client is to be used for internal API calls (CMDs and such)
 // The result HTTPRequestDoer can be supplied to OpenAPI generated clients for executing requests.
 // This does not use the generated client options for e.g. authentication,
 // because each generated OpenAPI client reimplements the client options using structs,
 // which makes them incompatible with each other, making it impossible to use write generic client code for common traits like authorization.
 // If the given authorization token builder is non-nil, it calls it and passes the resulting token as bearer token with requests.
-func CreateHTTPClient(cfg ClientConfig, generator AuthorizationTokenGenerator) (HTTPRequestDoer, error) {
+func CreateHTTPInternalClient(cfg ClientConfig, generator AuthorizationTokenGenerator) (HTTPRequestDoer, error) {
```

Comment:
a bit weird that this is under `core` and all the other under `http` like the rest. Import cycle on config?

Reply:
these are for our commands, the other are outgoing.